### PR TITLE
Add `SharedUnionFields` type

### PR DESCRIPTION
**Pull request by Emiyaaaaa**
Simple version of `SharedUnionFieldsDeep`

### CODE REVIEW

**Comment by Emiyaaaaa** [comment]
Hi @sindresorhus, would it be possible to add me to the funding? Thanks a lot!

**Comment by sindresorhus** [comment]
Of course: https://github.com/sindresorhus/type-fest/commit/ca449dc88984aff5b9691214177b0f61e4a172cd

**Review by som-sm — COMMENTED** [review]
@Emiyaaaaa Was going through this PR, and I have shared some thoughts. Let me know what you think.

File: source/shared-union-fields.d.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+@category Union
+*/
+export type SharedUnionFields<Union> =
+// If `Union` is not a union type, return `Union` directly.
+IsUnion<Union> extends false
+	? Union
+	// `Union extends` will convert `Union`
+	// to a [distributive conditionaltype](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html#distributive-conditional-types).
+	// But this is not what we want, so we need to wrap `Union` with `[]` to prevent it.
+	: [Union] extends [NonRecursiveType | ReadonlyMap<unknown, unknown> | ReadonlySet<unknown> | UnknownArray]
```
[the remark this answers is not in the document]
Comment (Emiyaaaaa) [reply source/shared-union-fields.d.ts:72]: > 2. Or, we can leave `RegExp` as is, and resolve the remaining members
>    ```ts
>    RegExp | {test: 1 | 2}
>    ```

This is better

**Pull request merged**